packages/react/src/Stack/Stack.tsx

```diff
@@ -0,0 +1,544 @@
+      padding: var(--stack-padding-condensed, 8px);
+    }
+
+    &[data-padding-wide='normal'] {
+      padding: var(--stack-padding-normal, 16px);
+    }
+
+    &[data-padding-wide='spacious'] {
+      padding: var(--stack-padding-spacious, 24px);
+    }
+
+    &[data-orientation-wide='horizontal'] {
+      flex-flow: row;
+    }
+
+    &[data-orientation-wide='vertical'] {
+      flex-flow: column;
+    }
+
+    &[data-gap-wide='none'] {
+      --Stack-gap-whenWide: 0;
+    }
+
+    &[data-gap-wide='condensed'] {
+      --Stack-gap-whenWide: var(--stack-gap-condensed, 8px);
+    }
+
+    &[data-gap-wide='normal'] {
+      --Stack-gap-whenWide: var(--stack-gap-normal, 16px);
+    }
+
+    &[data-gap-wide='spacious'] {
+      --Stack-gap-whenWide: var(--stack-gap-spacious, 24px);
+    }
+
+    &[data-align-wide='start'] {
+      align-items: flex-start;
+    }
+
+    &[data-align-wide='center'] {
+      align-items: center;
+    }
+
+    &[data-align-wide='end'] {
+      align-items: flex-end;
+    }
+
+    &[data-align-wide='baseline'] {
+      align-items: baseline;
+    }
+
+    &[data-spread-wide='start'] {
+      justify-content: flex-start;
+    }
+
+    &[data-spread-wide='center'] {
+      justify-content: center;
+    }
+
+    &[data-spread-wide='end'] {
+      justify-content: flex-end;
+    }
+
+    &[data-spread-wide='distribute'] {
+      justify-content: space-between;
+    }
+
+    &[data-spread-wide='distributeEvenly'] {
+      justify-content: space-evenly;
+    }
+  }
+`
+
+type GapScale = 'none' | 'condensed' | 'normal' | 'spacious'
+type Gap = GapScale | ResponsiveValue<GapScale>
+
+type OrientationScale = 'horizontal' | 'vertical'
+type Orientation = OrientationScale | ResponsiveValue<OrientationScale>
+
+type AlignScale = 'stretch' | 'start' | 'center' | 'end' | 'baseline'
+type Align = AlignScale | ResponsiveValue<AlignScale>
+
+type WrapScale = 'wrap' | 'nowrap'
+type Wrap = WrapScale | ResponsiveValue<WrapScale>
+
+type SpreadScale = 'start' | 'center' | 'end' | 'distribute' | 'distributeEvenly'
+type Spread = SpreadScale | ResponsiveValue<SpreadScale>
+
+type PaddingScale = 'none' | 'condensed' | 'normal' | 'spacious'
+type Padding = PaddingScale | ResponsiveValue<PaddingScale>
+
+type StackProps<As> = React.PropsWithChildren<{
+  /**
+   * Customize the element type of the rendered container
+   */
+  as?: As
+
+  /**
+   * Specify the gap between children elements in the stack
+   */
+  gap?: Gap
+
+  /**
+   * Specify the orientation for the stack container
+   * @default vertical
+   */
+  orientation?: Orientation
+
+  /**
+   * Specify the alignment between items in the cross-axis of the orientation
+   * @default stretch
+   */
+  align?: Align
+
+  /**
+   * Sets whether items are forced onto one line or can wrap onto multiple lines
+   * @default nowrap
+   */
+  wrap?: Wrap
+
+  /**
+   * Sets how items will be distributed in the stacking direction
+   * @default start
+   */
+  spread?: Spread
```

Comment:
Should we call this `justify` instead of spread to match what you'd expect it to be with css?